internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go

```diff
@@ -435,7 +436,7 @@ func (r ThreatIntelligenceIndicator) Create() sdk.ResourceFunc {
 			props.Pattern = &patternValue
 
 			if model.Confidence != -1 {
-				props.Confidence = utils.Int32(model.Confidence)
+				props.Confidence = pointer.To(int32(model.Confidence))
```

Comment:
```suggestion
				props.Confidence = pointer.To(int64(model.Confidence))
```

Reply:
This is a Kermit API, so `properties.Confidence` is actually 32-bit in the API for now until it's moved over to `go-azure-sdk`.